src/throttler.guard.ts

```diff
@@ -112,13 +117,25 @@ export class ThrottlerGuard implements CanActivate {
       // Check if specific limits are set at class or route level, otherwise use global options.
       const limit = await this.resolveValue(context, routeOrClassLimit || namedThrottler.limit);
       const ttl = await this.resolveValue(context, routeOrClassTtl || namedThrottler.ttl);
+      const blockDuration =
+        (await this.resolveValue(
+          context,
+          routeOrClassBlockDuration || namedThrottler.blockDuration,
+        )) || ttl;
```

Comment:
```suggestion
      const blockDuration =
        await this.resolveValue(
          context,
          routeOrClassBlockDuration || namedThrottler.blockDuration || ttl,
        );
```
From what I can see, no reason to not include the `ttl` as the fallback inside the `resolveValue` method

Reply:
Unless it's to handle the case of `routeOrClassBlockDuration` returning a falsey value (should `0` be allowed here) and then falling back to the `ttl`. That could make sense, and if so then ignore this suggestion